### ISSUE TITLE
Using `client.smartCollectionSync` defaults to wrong type declaration for `detailedResult` (always Falsy) vs using `smartCollectionSync` independently

### ISSUE THREAD

**Issue by garshythoel** [natelindev/tsdav#201]
For my code here:
```
const {
          created: createdObjects,
          updated: updatedObjects,
          deleted: deletedObjects
        } = (
          await client.smartCollectionSync({
            collection: {
              url: calendar.url,
              ctag: calendar.ctag,
              syncToken: calendar.syncToken,
              objects: localObjects.map(dbEventToDAVObject),
              objectMultiGet: client.calendarMultiGet
            },
            method: 'webdav',
            detailedResult: true
          })
        ).objects;
```

The TS typechecker throws error since it expects `detailedResult` to always be `false` (image below)
![CleanShot 2024-07-26 at 10 19 33@2x](https://github.com/user-attachments/assets/f49391a7-41c8-4dad-b86f-a6257b3c230d)

This also leads to the typechecker picking up the wrong overload for the function returns type and consequently errors on accessing `created`, `updated` and `deleted`

When i use `smartCollectionSync` directly (not from the clientMethod) - the overload for `detailedResult: true` is successfully picked up and no error

![CleanShot 2024-07-26 at 10 21 29@2x](https://github.com/user-attachments/assets/771caf99-f79e-4928-8a6b-100ed6c4a397)

I also get the same issues using `syncCalendars` independently vs from the `DAVClient` method.

I can use the methods directly but since the examples in the documentation are all DAVClient based probably a good idea to fix the overloads

**Comment by natelindev** [comment]
will take a look this week.

**Comment by natelindev** [comment]
fixed in v2.1.1
